### PR TITLE
campaigns: Don't use global database anymore

### PR DESCRIPTION
**Pull request by eseliger**
Since using global stores is discouraged, I've switched usage of those to be derived from out store.

### CODE REVIEW

File: enterprise/internal/campaigns/store/store.go [hunk 1]
```diff
@@ -74,6 +75,21 @@ func (s *Store) Transact(ctx context.Context) (*Store, error) {
...
+// ExternalServices returns a database.ExternalServiceStore using the same connection as this store.
+func (s *Store) ExternalServices() *database.ExternalServiceStore {
+	return database.ExternalServicesWith(s)
+}
+
+// UserCredentials returns a database.UserCredentialsStore using the same connection as this store.
+func (s *Store) UserCredentials() *database.UserCredentialsStore {
+	return database.UserCredentialsWith(s)
+}
+
```
Comment (LawnGnome) [inline enterprise/internal/campaigns/store/store.go:92]: I like these wrappers. Very clean.

**Review by LawnGnome — APPROVED** [review]
This is actually way cleaner than I would have expected. Nice work!

File: enterprise/internal/campaigns/reconciler/executor.go [hunk 2]
```diff
@@ -607,28 +610,19 @@ func loadCampaign(ctx context.Context, tx getCampaigner, id int64) (*campaigns.C
...
-		UserID:              userID,
-		ExternalServiceType: repo.ExternalRepo.ServiceType,
-		ExternalServiceID:   repo.ExternalRepo.ServiceID,
-	})
+type getNamespacer interface {
+	GetByID(ctx context.Context, orgID, userID int32) (*database.Namespace, error)
 }
 
-func decorateChangesetBody(ctx context.Context, tx getCampaigner, cs *repos.Changeset) error {
+func decorateChangesetBody(ctx context.Context, tx getCampaigner, nsStore getNamespacer, cs *repos.Changeset) error {
```
[the remark this answers is not in the document]
Comment (eseliger) [reply enterprise/internal/campaigns/reconciler/executor.go:617]: I know, but I wanted to keep an interface so when we want to write a test for this function we have the interface at hand to implement some `mockNamespaceStore`. Does that make sense or is this against the Go way of life ™️ ?

[the remark this answers is not in the document]
Comment (LawnGnome) [reply enterprise/internal/campaigns/reconciler/executor.go:617]: I think it's totally fine: it's more readable, and wanting to reuse it later for tests makes total sense. I just didn't know if you knew about the inline `interface {}` functionality in general!

**Review by mrnugget — APPROVED** [review]
Nice!

**Pull request merged**